lib/chartmogul/config_attributes.rb

```diff
@@ -4,17 +4,33 @@ module ChartMogul
   module ConfigAttributes
     def config_accessor(attribute, default_value = nil)
       define_method(attribute) do
-        attr = config.send(attribute) || default_value
+        attr = config.send(attribute) || global_config.send(attribute) || default_value
         if attr.nil?
           raise ConfigurationError, "Configuration for #{attribute} not set"
         end
 
         attr
       end
 
+      define_method("global_#{attribute}") do
```

Comment:
The library itself uses the variant that looks into the thread-local configuration first.
This method is for feature-completion, in case you somehow wanted to have a look at it.
Yes, it will raise, but that would be expected I think, considering how the non-global methods behave.